reactive/etcd.py

```diff
@@ -663,15 +663,21 @@ def force_rejoin_requested():
     check_cluster_health()
 
 
-@hook("cluster-relation-broken")
+@hook("stop")
 def perform_self_unregistration(cluster=None):
     """Attempt self removal during unit teardown."""
     etcdctl = EtcdCtl()
     leader_address = leader_get("leader_address")
     unit_name = os.getenv("JUJU_UNIT_NAME").replace("/", "")
     members = etcdctl.member_list()
     # Self Unregistration
-    etcdctl.unregister(members[unit_name]["unit_id"], leader_address)
+    for _ in range(5):
+        try:
+            etcdctl.unregister(members[unit_name]["unit_id"], leader_address)
+            break
+        except EtcdCtl.CommandFailed:
+            # Randomized back-off timer to let cluster settle
+            time.sleep(random.randint(1, 3))
```

Comment:
As with previous loop comments, shouldn't we log loop errors here?  If we truly don't care about the `unregister` return (because we're in a stop hook), why loop?  If we do, let's log what we're retrying with a final message and/or raise if we reach the end of our loop.